pkg/cloudprovider/gce/gce.go

```diff
@@ -628,8 +630,16 @@ func (gce *GCECloud) NodeAddresses(_ string) ([]api.NodeAddress, error) {
 	}, nil
 }
 
+func (gce *GCECloud) isCurrentInstance(instance string) bool {
+	return gce.instanceID == canonicalizeInstanceName(instance)
+}
+
 // ExternalID returns the cloud provider ID of the specified instance (deprecated).
 func (gce *GCECloud) ExternalID(instance string) (string, error) {
+	// if we are asking about the current instance, just go to metadata
+	if gce.isCurrentInstance(instance) {
+		return gce.getCurrentExternalID()
```

Comment:
We get the instanceID and network name once (when initializing the cloud provider). Is there a reason not to grab the external ID at the same time and just hold it in the gce struct as another variable?